server/src/main/java/com/linecorp/centraldogma/server/PluginGroup.java

```diff
@@ -63,42 +67,60 @@ final class PluginGroup {
      *
      * @param target the {@link PluginTarget} which would be loaded
      */
+    @VisibleForTesting
     @Nullable
     static PluginGroup loadPlugins(PluginTarget target, CentralDogmaConfig config) {
-        return loadPlugins(PluginGroup.class.getClassLoader(), target, config, ImmutableList.of());
+        return loadPlugins(PluginGroup.class.getClassLoader(), config, ImmutableList.of()).get(target);
     }
 
     /**
      * Returns a new {@link PluginGroup} which holds the {@link Plugin}s loaded from the classpath.
-     * {@code null} is returned if there is no {@link Plugin} whose target equals to the specified
+     * An empty map is returned if there is no {@link Plugin} whose target equals to the specified
      * {@code target}.
      *
      * @param classLoader which is used to load the {@link Plugin}s
-     * @param target the {@link PluginTarget} which would be loaded
      */
-    @Nullable
-    static PluginGroup loadPlugins(ClassLoader classLoader, PluginTarget target, CentralDogmaConfig config,
-                                   List<Plugin> plugins) {
+    static Map<PluginTarget, PluginGroup> loadPlugins(ClassLoader classLoader, CentralDogmaConfig config,
```

Comment:
Note: Understood that with this change, we don't allow the same plugin to be added with different `PluginTarget`s (i.e. `PluginA+LEADER_ONLY`, `PluginA+ZONE_LEADER_ONLY`)